raspyrfm_client/device/manufacturer/hama/_121938.py

```diff
@@ -0,0 +1,47 @@
+from raspyrfm_client.device import actions
+from raspyrfm_client.device.base import Device
+
+
+class HAMA(Device):
```

Comment:
OK! And: The generation of the connairstring "TX..." should not be done in base.py. base.by should only return the pulsedata- and parameters (repetitions, timebase...), generate_code should disappear at all.
Maybe having a gateway-baseclass which gets the pulse data from the device object and the gateway implementations generate the appropriate data? Ideas?